File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Gulf/GulfB_rd_geo.py
#==================================================================
#  
#  Bellhop: Topgulf, Range dep., Geometric beams 
#  Mexilhoeira Grande, Dom Jul 30 14:44:50 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run GulfB_rd_geo

from os import system
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("Bellhop - Topgulf, Range dep., Geometric beams:")
print("Coherent TL calculation")
print("Geometric hat beams in Cartesian coordinates")

fid = open('GulfB_rd_geo.bty','r')
theline = fid.readline()
theline = fid.readline()
n       = int( theline )
rbtykm  = zeros( n ) 
zbty    = zeros( n )
for i in range(n):
    theline = str( fid.readline() )
    datai = theline.split()
    rbtykm[ i] = float( datai[0] )
    zbty[   i] = float( datai[1] )
fid.close()

rbty = rbtykm*1000

system("bellhop.exe GulfB_rd_geo")

filename = 'GulfB_rd_geo.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )
tl[tl > 120.0] = 120.0

figure(1)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower',vmin=60,vmax=120)
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,zs,marker="<",markersize=16,color="k")
plot(rbtykm,zbty,'k')
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Topgulf, Range dep., Geometric beams')
xlim(0,rmaxkm)
ylim(Dmax,0)

show()

print("done.")

File: acoustics_toolbox_r2020/Python/Bellhop/Tests/DW/Gulf/GulfB_rd_geo.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/Plot
addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Bellhop - Topgulf, Range dep., Geometric beams:')
disp('Coherent TL calculation')
disp('Geometric hat beams in Cartesian coordinates') 

fid = fopen('GulfB_rd_geo.bty','r');
s = fscanf(fid,'%s\n',[1 1]);
n = fscanf(fid,'%d\n',[1 1]);
data = fscanf(fid,'%f %f\n',[2 n]);
fclose(fid);

rbtykm = data(1,:); rbty = rbtykm*1000; 
zbty   = data(2,:);

system('bellhop.exe GulfB_rd_geo');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'GulfB_rd_geo.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

figure(1), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), %caxis([40 90]), 
colorbar
plot(rbtykm,zbty,'k','LineWidth',2)
axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('Bellhop - Topgulf, Range dep., Geometric beams')

disp('done.')
